Changes grapple physics calculations

mothra42 opened the pull request:
Changes the grapple so it pulls stronger towards the hook creating more consistent physics. Also adds a new boolean flag that can be used to toggle a velocity visualizer: "should_show_trajectory_debug"

mothra42 committed:
Changes grapple physics calculations

Changes the grapple so it pulls stronger towards the hook creating more
consistent physics. Also adds a new boolean flag that can be used to
toggle a velocity visualizer: "should_show_trajectory_debug"

tiny-crab commented on Player/Player.gd:
```diff
@@ -170,15 +172,22 @@ func move_jump(delta):
+    hook_velocity = Vector2.ZERO
+    if velocity.y >= DOWNWARD_VELOCITY_LIMIT:
+        print("using enhanced grapple")
+        hook_velocity = to_local(grapplingHook.tip).normalized() * CHAIN_PULL * delta * 35
```
Is this pulling the player faster if they're at a high downward velocity to quickly compensate for changing direction?

mothra42 replied:
Yeah, exactly! I found that when falling at high enough speeds the grapple's pull was not effective enough to actually pull you up at all, and this doesn't feel fantastic. Boosting the grapple velocity allows the player to actually come out of their fall and potentially grapple to a land spot. Another solution is to cap our falling speed to allow players to react better.

tiny-crab commented on Player/Player.tscn:
```diff
@@ -209,7 +209,7 @@ position = Vector2( 0, -24 )
 hframes = 9
 vframes = 13
-frame = 97
+frame = 98
```
lol I hate that this file updates while the animation is playing in the editor. Wonder if we can have like gitignore properties for Nodes.

mothra42 replied:
That would be nice. We can't ignore a .tscn file entirely though because we may add new nodes that need to be tracked.

tiny-crab replied:
Yeah :( it seems like we'd have to somehow tell Godot to stop tracking certain things in the version control tscn file. Oh well 🤷

mothra42 committed:
Change velocity on grapple hook release

Adds a little extra velocity based on the dot product between the
velocity vector and the vector point from the player to the grapple
position.
